### PR TITLE
Add first attempt at UCP API conventions for review

### PR DESCRIPTION
**Pull request by bryan-strassner**

### CODE REVIEW

File: docs/api-conventions.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Headers
+### Required
+
+* X-Auth-Token  
+The auth token to identify the invoking user.
+
+### Optional
+
+* X-Context-Marker  
+A context id that will be carried on all logs for this cleint-provided marker. This marker may only be a 36-character canonical representation of an UUID  (8-4-4-4-12)
```
Comment (sh8121att) [inline docs/api-conventions.md:43]: clieint->client

File: docs/api-conventions.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/api/v1.0/sampleresources/ExTeRnAlNAME-1234
+      ^         ^       ^       ^
+      |         |       |      defer to external naming
+      |         |      plural
+      |        lower case
+     version here
+```
+---
+## Error responses
+Error responses (HTTP response body accompanying 4xx series responses) are a specialization of the [Kubernetes standard for error representation](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#response-status-kind).
```
Comment (sh8121att) [inline docs/api-conventions.md:18]: Should this not also apply to 5XX status codes. Nothing better than getting a blank-bodied 500 back.

**Review by sh8121att — COMMENTED** [review]
This LGTM - just one question.

**Comment by mark-burnett** [comment]
Closing to enable migration to OpenStack

**Pull request closed without merging**